lib/gds_api/test_helpers/publishing_api.rb

```diff
@@ -643,10 +641,33 @@ def stub_publishing_api_unreserve_path_invalid(base_path, publishing_app = /.*/)
         stub_publishing_api_unreserve_path_with_code(base_path, publishing_app, 422)
       end
 
-      def stub_publishing_api_put_intent(base_path, body = intent_for_publishing_api(base_path))
-        url = PUBLISHING_API_ENDPOINT + "/publish-intent" + base_path
-        body = body.to_json unless body.is_a?(String)
-        stub_request(:put, url).with(body: body).to_return(status: 200, body: "{}", headers: { "Content-Type" => "application/json; charset=utf-8" })
+      # Stub a PUT /publish-intent/:base_path request with the given base_path
+      # and request body.
+      #
+      # @example
+      #   stub_publishing_api_put_intent(
+      #     "/path/to/content",
+      #     publishing_app: "publisher",
+      #     rendering_app: "frontend",
+      #     publish_time: "2019-11-11T17:56:17+00:00",
+      #   )
+      #
+      # @param base_path [String]
+      # @param params [Hash]
+      def stub_publishing_api_put_intent(base_path, params = {})
+        url = PUBLISHING_API_ENDPOINT + "/publish-intent#{base_path}"
+        body = if params.is_a?(String)
+                 params
+               else
+                 { base_path: base_path }.merge(params).to_json
```

Comment:
Why does `base_path` need to be merged here?

Reply:
Ah good catch - I was under the impression that Publishing API did return the payload with a base_path (as per [path_reservation](https://github.com/alphagov/gds-api-adapters/blob/6193489874ce1305ec093f1a0ff14712e440940f/lib/gds_api/test_helpers/publishing_api.rb#L710)) 

However because of the rather strange conditional in https://github.com/alphagov/publishing-api/blob/master/app/commands/put_publish_intent.rb#L6-L9 it won't actually include a base_path.